### DISCUSSION TITLE
GitHub App access to Packages

### DISCUSSION THREAD

**Discussion by mat007** [community/community#38287] [Packages]
Hi,

[Authenticating to GitHub Packages](https://docs.github.com/en/packages/working-with-a-github-packages-registry/working-with-the-npm-registry#authenticating-to-github-packages) says «GitHub Packages only supports authentication using a personal access token (classic).».
Does this mean a GitHub App token (ghs or ghu) can’t be used to read packages?

Here is what I did:
- changed the App permissions to add read-only to Packages
- accepted the change on the private repo the App is installed to
- retrieved both ghs and ghu tokens (I can detail a bit how I did that if needed)
- tested that the tokens could be used to access the private repo files and such
- used `curl` to list the packages:
```
curl -H "Accept: application/vnd.github+json" -H "Authorization: Bearer TOKEN" https://api.github.com/orgs/ORG/packages?package_type=npm
```
With a ghu token I’m getting
```
{                                                                                                                        
  "message": "Resource not accessible by integration",                                                                   
  "documentation_url": "https://docs.github.com/rest/reference/packages#list-packages-for-an-organization"               
}                                                                                                                        
```
and with a ghs token, an empty array
```
[  
   
]  
```
The array is not empty when using one of my own PAT.

Any suggestion for how I could make my App access the npm packages?
Cheers!

**Comment by david-engelmann** [comment]
You can add in ssh files and an agent to automatically authenticate without any prompts

**Comment by mat007** [comment]
According to #34324 it seems not possible at the moment…
